### PR TITLE
Switch tests and examples to vtgate discoverygateway.

### PR DESCRIPTION
**Pull request by enisoc**
@guoliang100

[Review on Reviewable](https://reviewable.io/reviews/youtube/vitess/1531)

### CODE REVIEW

File: test/config.json [hunk 1]
```diff
@@ -322,7 +322,9 @@
 		},
 		"vtgatev2": {
 			"File": "vtgatev2_test.py",
-			"Args": [],
+			"Args": [
+				"--vtgate-gateway-flavor=shardgateway"
```
Comment (guoliang100) [inline test/config.json]: Do you mean "discoverygateway" here?

Comment (enisoc) [inline test/config.json]: I changed the default value of --vtgate-gateway-flavor to discoverygateway, so I added an explicit value here for the one that still tests shardgateway. Internally I checked that we always explicitly specify --vtgate-gateway-flavor in our test configs.

**Comment by enisoc** [comment]
It looks like a lot of tests are still making assumptions about the specific way shardgateway works. I'll take a look at them tomorrow.

**Comment by enisoc** [comment]
@guoliang100 I had to update some tests to work with discoverygateway. Can you take another look? Thanks.

**Comment by guoliang100** [comment]
LGTM

**Pull request merged**